// @voker/voker/dist/chunk-N7WW272A.js
// src/schema/base/agent-versions.ts
var AgentVersionT = {
  deserialize: (data) => {
    return {
      agentVersionName: data.agent_version_name,
      description: data.description,
      agentVersionNumber: data.agent_version_number,
      createdAt: new Date(data.created_at),
      deprecatedAt: data.deprecated_at !== null ? new Date(data.deprecated_at) : null
    };
  }
};
var AgentVersionNotFoundError = class extends Error {
};

// src/schema/base/agents.ts
var AgentT = {
  deserialize: (data) => {
    return {
      agentName: data.agent_name,
      description: data.description,
      agentNumber: data.agent_number,
      createdAt: new Date(data.created_at),
      deprecatedAt: data.deprecated_at !== null ? new Date(data.deprecated_at) : null
    };
  }
};
var AgentNotFoundError = class extends Error {
};

// src/schema/base/people.ts
var PersonT = {
  deserialize: (data) => {
    return {
      personId: data.person_id,
      properties: data.properties,
      createdAt: new Date(data.created_at)
    };
  }
};
var PersonNotFoundError = class extends Error {
};

// src/schema/base/fingerprints.ts
var FingerprintT = {
  deserialize: (data) => ({
    fingerprintId: data.fingerprint_id
  })
};
var FingerprintNotFoundError = class extends Error {
};

export {
  AgentVersionT,
  AgentVersionNotFoundError,
  AgentT,
  AgentNotFoundError,
  PersonT,
  PersonNotFoundError,
  FingerprintT,
  FingerprintNotFoundError
};
