// @voker/voker/dist/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  ApiClient: () => ApiClient,
  VokerClient: () => VokerClient
});
module.exports = __toCommonJS(index_exports);

// src/schema/api/agent-versions.ts
var CreateAgentVersionPayloadT = {
  serialize: (data) => {
    return {
      agent_version_name: data.agentVersionName,
      description: data.description
    };
  }
};

// src/schema/api/agents.ts
var CreateAgentPayloadT = {
  serialize: (data) => {
    return {
      agent_name: data.agentName,
      description: data.description
    };
  }
};

// src/schema/api/events.ts
var CreateEventPayloadT = {
  serialize: (data) => {
    return {
      event_name: data.eventName,
      properties: data.properties,
      fingerprint_id: data.fingerprintId,
      agent: data.vokerAgent,
      agent_version: data.vokerAgentVersion,
      person: data.vokerPerson,
      session: data.vokerSession
    };
  }
};

// src/schema/api/fingerprints.ts
var CreateFingerprintPayloadT = {
  serialize: (data) => {
    return {
      language: data.language,
      language_version: data.languageVersion,
      system: data.system,
      openai_sdk_version: data.openaiSdkVersion,
      anthropic_sdk_version: data.anthropicSdkVersion,
      gemini_sdk_version: data.geminiSdkVersion,
      voker_sdk_version: data.vokerSdkVersion,
      all_packages: data.allPackages,
      git_repository_url: data.gitRepositoryUrl,
      git_branch: data.gitBranch,
      git_commit_hash: data.gitCommitHash
    };
  }
};

// src/schema/api/people.ts
var CreatePersonPayloadT = {
  serialize: (data) => {
    return {
      person_id: data.personId,
      properties: data.properties
    };
  }
};
var UpdatePersonPayloadT = {
  serialize: (data) => {
    return {
      properties: data.properties
    };
  }
};

// src/schema/base/agent-versions.ts
var AgentVersionT = {
  deserialize: (data) => {
    return {
      agentVersionName: data.agent_version_name,
      description: data.description,
      agentVersionNumber: data.agent_version_number,
      createdAt: new Date(data.created_at),
      deprecatedAt: data.deprecated_at !== null ? new Date(data.deprecated_at) : null
    };
  }
};
var AgentVersionNotFoundError = class extends Error {
};

// src/schema/base/agents.ts
var AgentT = {
  deserialize: (data) => {
    return {
      agentName: data.agent_name,
      description: data.description,
      agentNumber: data.agent_number,
      createdAt: new Date(data.created_at),
      deprecatedAt: data.deprecated_at !== null ? new Date(data.deprecated_at) : null
    };
  }
};
var AgentNotFoundError = class extends Error {
};

// src/schema/base/people.ts
var PersonT = {
  deserialize: (data) => {
    return {
      personId: data.person_id,
      properties: data.properties,
      createdAt: new Date(data.created_at)
    };
  }
};
var PersonNotFoundError = class extends Error {
};

// src/schema/base/fingerprints.ts
var FingerprintT = {
  deserialize: (data) => ({
    fingerprintId: data.fingerprint_id
  })
};
var FingerprintNotFoundError = class extends Error {
};

// src/api/client.ts
var import_os = __toESM(require("os"), 1);
var import_fs = __toESM(require("fs"), 1);
var import_path = __toESM(require("path"), 1);
var import_child_process = require("child_process");
var ApiClient = class {
  constructor(config = {
    apiKey: process.env.VOKER_API_KEY,
    baseUrl: "https://evals.voker.ai"
  }) {
    this.config = config;
    if (config.apiKey === void 0) {
      throw new Error(
        "Missing credentials. Please set the `VOKER_API_KEY` environment variable."
      );
    }
  }
  /**
   * Access [events](https://app.voker.ai/docs/concepts/events#events) related endpoints.
   */
  get events() {
    return new ApiClient_Events(this);
  }
  /**
   * Access [people](https://app.voker.ai/docs/concepts/people) related endpoints.
   */
  get people() {
    return new ApiClient_People(this);
  }
  /**
   * Access [fingerprints](https://app.voker.ai/docs/concepts/events#fingerprints) related endpoints.
   */
  get fingerprints() {
    return new ApiClient_Fingerprints(this);
  }
  /**
   * Access [agents](https://app.voker.ai/docs/concepts/agents#agents) related endpoints.
   */
  get agents() {
    return new ApiClient_Agents(this);
  }
  /**
   * Access [agent versions](https://app.voker.ai/docs/concepts/agents#agent-versions) related endpoints.
   */
  get agentVersions() {
    return new ApiClient_AgentVersions(this);
  }
  async request(method, path2, body) {
    const req = await fetch(`${this.config.baseUrl}${path2}`, {
      method,
      headers: {
        "Content-Type": "application/json",
        Authorization: `Bearer ${this.config.apiKey}`
      },
      body: body !== void 0 ? JSON.stringify(body) : void 0
    });
    return req;
  }
};
var ApiClient_BaseSubPath = class {
  constructor(vokerClient) {
    this.vokerClient = vokerClient;
  }
};
var ApiClient_Events = class extends ApiClient_BaseSubPath {
  /**
   * Create a new [event](https://app.voker.ai/docs/concepts/events).
   *
   * See [the reference documentation](https://app.voker.ai/docs/integrations/sdks/typescript#voker.src.api.client.ApiClient_Events.create)
   */
  async create(payload) {
    const response = await this.vokerClient.request(
      "POST",
      "/api/v1/events",
      CreateEventPayloadT.serialize(payload)
    );
    const data = await response.json();
    if (response.status == 404) {
      const error = data;
      throw new FingerprintNotFoundError();
    }
    if (!response.ok) {
      throw new Error("something went wrong");
    }
  }
};
var ApiClient_People = class extends ApiClient_BaseSubPath {
  /**
   * Get a person.
   */
  async get(personId) {
    const response = await this.vokerClient.request(
      "GET",
      `/api/v1/people/${personId}`
    );
    const data = await response.json();
    if (response.status == 404) {
      const error = data;
      throw new PersonNotFoundError();
    }
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    return PersonT.deserialize(data);
  }
  /**
   * Create a person.
   *
   * See [Creating a Person](https://app.voker.ai/docs/concepts/people#creating-a-person) for more information.
   */
  async create(payload) {
    const response = await this.vokerClient.request(
      "PUT",
      "/api/v1/people",
      CreatePersonPayloadT.serialize(payload)
    );
    const data = await response.json();
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    return PersonT.deserialize(data);
  }
  /**
   * Update a person.
   */
  async update(personId, payload) {
    const response = await this.vokerClient.request(
      "PUT",
      `/api/v1/people/${personId}`,
      UpdatePersonPayloadT.serialize(payload)
    );
    const data = await response.json();
    if (response.status == 404) {
      const error = data;
      throw new PersonNotFoundError();
    }
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    return PersonT.deserialize(data);
  }
};
var safeGit = (cmd) => {
  try {
    return (0, import_child_process.execSync)(cmd, { stdio: ["ignore", "pipe", "ignore"] }).toString().trim();
  } catch {
    return void 0;
  }
};
var ApiClient_Fingerprints = class extends ApiClient_BaseSubPath {
  /**
   * Create a fingerprint.
   *
   * See [Creating a Fingerprint](https://app.voker.ai/docs/concepts/events#creating-a-fingerprint) for more information.
   */
  async create() {
    const payload = {
      language: "javascript",
      languageVersion: process.version,
      system: `${import_os.default.platform()} ${import_os.default.release()} ${import_os.default.arch()}`
    };
    try {
      const rootPkgPath = import_path.default.resolve(process.cwd(), "package.json");
      const rootPkg = JSON.parse(import_fs.default.readFileSync(rootPkgPath, "utf-8"));
      const allPackages = {
        ...rootPkg.dependencies ?? {},
        ...rootPkg.devDependencies ?? {},
        ...rootPkg.peerDependencies ?? {}
      };
      payload.allPackages = allPackages;
      payload.openaiSdkVersion = allPackages.openai;
      payload.anthropicSdkVersion = allPackages["@anthropic-ai/sdk"];
      payload.geminiSdkVersion = allPackages["@google/generative-ai"];
      payload.vokerSdkVersion = allPackages.voker;
    } catch {
    }
    payload.gitRepositoryUrl = safeGit(
      "git config --get remote.origin.url"
    );
    payload.gitBranch = safeGit("git rev-parse --abbrev-ref HEAD");
    payload.gitCommitHash = safeGit("git rev-parse HEAD");
    payload.gitCommitDate = safeGit("git show -s --format=%ci HEAD");
    const response = await this.vokerClient.request(
      "PUT",
      "/api/v1/fingerprints",
      CreateFingerprintPayloadT.serialize(payload)
    );
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    const data = await response.json();
    return FingerprintT.deserialize(data);
  }
};
var ApiClient_Agents = class extends ApiClient_BaseSubPath {
  /**
   * Create an agent.
   *
   * See [Creating an Agent](https://app.voker.ai/docs/concepts/agents#creating-agent) for more information.
   */
  async create(payload) {
    const response = await this.vokerClient.request(
      "PUT",
      "/api/v1/agents",
      CreateAgentPayloadT.serialize(payload)
    );
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    const data = await response.json();
    return AgentT.deserialize(data);
  }
  /**
   * Get an agent.
   */
  async get(agentName) {
    const response = await this.vokerClient.request(
      "GET",
      `/api/v1/agents/${agentName}`
    );
    const data = await response.json();
    if (response.status == 404) {
      const error = data;
      throw new AgentNotFoundError();
    }
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    return AgentT.deserialize(data);
  }
};
var ApiClient_AgentVersions = class extends ApiClient_BaseSubPath {
  /**
   * Create an agent version.
   *
   * See [Creating an Agent Version](https://app.voker.ai/docs/concepts/agents#creating-agent-version) for more information.
   */
  async create(agentName, payload) {
    const response = await this.vokerClient.request(
      "PUT",
      `/api/v1/agents/${agentName}/agent-versions`,
      CreateAgentVersionPayloadT.serialize(payload)
    );
    const data = await response.json();
    if (response.status == 404) {
      const error = data;
      throw new AgentNotFoundError();
    }
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    return AgentVersionT.deserialize(data);
  }
  /**
   * Get an agent version.
   */
  async get(agentName, agentVersionName) {
    const response = await this.vokerClient.request(
      "GET",
      `/api/v1/agents/${agentName}/agent-versions/${agentVersionName}`
    );
    const data = await response.json();
    if (response.status == 404) {
      const error = data;
      if (error.detail.type === "agent-not-found") {
        throw new AgentNotFoundError();
      }
      if (error.detail.type === "agent-version-not-found") {
        throw new AgentVersionNotFoundError();
      }
    }
    if (!response.ok) {
      throw new Error("something went wrong");
    }
    return AgentVersionT.deserialize(data);
  }
};

// src/api/batch.ts
var EventWorker = class {
  constructor(apiClient) {
    this.apiClient = apiClient;
    this.requests = [];
    this.start().catch(console.error);
  }
  requests;
  running = false;
  intervalMs = 1e3;
  timeout = null;
  fingerprint = null;
  startPromise = null;
  addRequest(item) {
    this.requests.push(item);
  }
  async start() {
    if (this.startPromise !== null) {
      return this.startPromise;
    }
    this.startPromise = (async () => {
      this.fingerprint = await this.apiClient.fingerprints.create();
      this.timeout = setTimeout(
        this.process.bind(this),
        this.intervalMs
      ).unref();
      this.running = true;
    })();
    return this.startPromise;
  }
  async close() {
    if (this.startPromise !== null) {
      await this.startPromise;
    }
    this.running = false;
    if (this.timeout !== null) {
      clearTimeout(this.timeout);
      this.timeout = null;
    }
    if (this.requests.length > 0) {
      await this.process();
    }
  }
  async process() {
    const fingerprint = this.fingerprint;
    if (fingerprint === null) {
      throw new Error("Fingerprint is not set");
    }
    const requestsToProcess = [...this.requests];
    this.requests = [];
    for (const request of requestsToProcess) {
      await this.processOne(request, fingerprint);
    }
    if (!this.running) {
      return;
    }
    this.timeout = setTimeout(
      this.process.bind(this),
      this.intervalMs
    ).unref();
  }
  async processOne(item, fingerprint) {
    if (item.path === "events") {
      if (item.operation === "create") {
        const payload = item.payload;
        await this.apiClient.events.create({
          ...payload,
          fingerprintId: fingerprint.fingerprintId
        });
      }
    } else if (item.path === "people") {
      if (item.operation === "create") {
        await this.apiClient.people.create(item.payload);
      } else if (item.operation === "update") {
        await this.apiClient.people.update(
          item.params["personId"],
          item.payload
        );
      }
    } else if (item.path === "agents") {
      if (item.operation === "create") {
        await this.apiClient.agents.create(item.payload);
      }
    } else if (item.path === "agent_versions") {
      if (item.operation === "create") {
        await this.apiClient.agentVersions.create(
          item.params["agentName"],
          item.payload
        );
      }
    }
  }
};

// src/client.ts
var VokerClient = class {
  eventWorker;
  constructor(config) {
    const eventWorker = new EventWorker(new ApiClient(config));
    this.eventWorker = eventWorker;
    const closeHandler = () => {
      this.close().catch(console.error);
    };
    const signalHandler = (signal) => {
      this.close().catch(console.error).finally(() => process.kill(process.pid, signal));
    };
    process.once("beforeExit", closeHandler);
    process.once("SIGTERM", () => signalHandler("SIGTERM"));
    process.once("SIGINT", () => signalHandler("SIGINT"));
  }
  /**
   * Start the client. This will start the background worker that sends events to the Voker API.
   */
  async start() {
    await this.eventWorker.start();
  }
  /**
   * Close the client. This will stop the background worker and flush any remaining events to the Voker API.
   */
  async close() {
    await this.eventWorker.close();
  }
  /**
   * Access [events](https://app.voker.ai/docs/concepts/events#events) related endpoints.
   */
  get events() {
    return new VokerClient_Events(this.eventWorker);
  }
  /**
   * Access [people](https://app.voker.ai/docs/concepts/people) related endpoints.
   */
  get people() {
    return new VokerClient_People(this.eventWorker);
  }
  /**
   * Access [agents](https://app.voker.ai/docs/concepts/agents#agents) related endpoints.
   */
  get agents() {
    return new VokerClient_Agents(this.eventWorker);
  }
  /**
   * Access [agent versions](https://app.voker.ai/docs/concepts/agents#agent-versions) related endpoints.
   */
  get agentVersions() {
    return new VokerClient_AgentVersions(this.eventWorker);
  }
};
var VokerClient_Subpath = class {
  constructor(eventWorker) {
    this.eventWorker = eventWorker;
  }
};
var VokerClient_Events = class extends VokerClient_Subpath {
  /**
   * Create a new [event](https://app.voker.ai/docs/concepts/events).
   *
   * See [the reference documentation](https://app.voker.ai/docs/integrations/sdks/typescript#voker.src.client.VokerClient_Events.create)
   */
  create(payload) {
    const req = {
      path: "events",
      operation: "create",
      params: {},
      payload
    };
    this.eventWorker.addRequest(req);
  }
};
var VokerClient_People = class extends VokerClient_Subpath {
  /**
   * Create a person.
   *
   * See [Creating a Person](https://app.voker.ai/docs/concepts/people#creating-a-person) for more information.
   */
  create(payload) {
    const req = {
      path: "people",
      operation: "create",
      params: {},
      payload
    };
    this.eventWorker.addRequest(req);
  }
  /**
   * Update a person.
   */
  update(personId, payload) {
    const req = {
      path: "people",
      operation: "update",
      params: { personId },
      payload
    };
    this.eventWorker.addRequest(req);
  }
};
var VokerClient_Agents = class extends VokerClient_Subpath {
  /**
   * Create an agent.
   *
   * See [Creating an Agent](https://app.voker.ai/docs/concepts/agents#creating-agent) for more information.
   */
  create(payload) {
    const req = {
      path: "agents",
      operation: "create",
      params: {},
      payload
    };
    this.eventWorker.addRequest(req);
  }
};
var VokerClient_AgentVersions = class extends VokerClient_Subpath {
  /**
   * Create an agent version.
   *
   * See [Creating an Agent Version](https://app.voker.ai/docs/concepts/agents#creating-agent-version) for more information.
   */
  create(agentName, payload) {
    const req = {
      path: "agent_versions",
      operation: "create",
      params: { agentName },
      payload
    };
    this.eventWorker.addRequest(req);
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ApiClient,
  VokerClient
});
