// @voker/voker/dist/ai/provider-aisdk/index.js
import {
  VokerClient,
  checkLlmEventParams
} from "../../chunk-CI7WS3DE.js";
import "../../chunk-N7WW272A.js";
import "../../chunk-EAQYK3U2.js";

// src/ai/provider-aisdk/index.ts
var wrapAiSdk = (baseAi, vokerClient) => {
  const { generateText: baseGenerateText, ...rest } = baseAi;
  if (vokerClient === void 0) {
    vokerClient = new VokerClient();
  }
  return {
    ...rest,
    generateText: wrapGenerateText(baseGenerateText, vokerClient)
  };
};
var wrapGenerateText = (baseGenerateText, vokerClient) => {
  return async (params) => {
    const {
      vokerAgent,
      vokerAgentVersion,
      vokerPerson,
      vokerSession,
      ...restParams
    } = params;
    const vokerParams = checkLlmEventParams({
      vokerAgent,
      vokerAgentVersion,
      vokerPerson,
      vokerSession
    });
    const result = await baseGenerateText(restParams);
    const providerNames = {
      "anthropic.messages": "anthropic-messages",
      "google.generative-ai": "gemini-models",
      "openai.responses": "openai-responses"
    };
    for (const step of result.steps) {
      const providerName = providerNames[step.model.provider];
      if (providerName === void 0) {
        continue;
      }
      vokerClient.events.create({
        eventName: "llm",
        ...vokerParams,
        properties: {
          api: providerName,
          inputs: step.request.body,
          output: step.response.body
        }
      });
    }
    return result;
  };
};
export {
  wrapAiSdk
};
