// @voker/voker/dist/schema/base/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/schema/base/index.ts
var base_exports = {};
__export(base_exports, {
  AgentNotFoundError: () => AgentNotFoundError,
  AgentT: () => AgentT,
  AgentVersionNotFoundError: () => AgentVersionNotFoundError,
  AgentVersionT: () => AgentVersionT,
  FingerprintNotFoundError: () => FingerprintNotFoundError,
  FingerprintT: () => FingerprintT,
  PersonNotFoundError: () => PersonNotFoundError,
  PersonT: () => PersonT
});
module.exports = __toCommonJS(base_exports);

// src/schema/base/agent-versions.ts
var AgentVersionT = {
  deserialize: (data) => {
    return {
      agentVersionName: data.agent_version_name,
      description: data.description,
      agentVersionNumber: data.agent_version_number,
      createdAt: new Date(data.created_at),
      deprecatedAt: data.deprecated_at !== null ? new Date(data.deprecated_at) : null
    };
  }
};
var AgentVersionNotFoundError = class extends Error {
};

// src/schema/base/agents.ts
var AgentT = {
  deserialize: (data) => {
    return {
      agentName: data.agent_name,
      description: data.description,
      agentNumber: data.agent_number,
      createdAt: new Date(data.created_at),
      deprecatedAt: data.deprecated_at !== null ? new Date(data.deprecated_at) : null
    };
  }
};
var AgentNotFoundError = class extends Error {
};

// src/schema/base/people.ts
var PersonT = {
  deserialize: (data) => {
    return {
      personId: data.person_id,
      properties: data.properties,
      createdAt: new Date(data.created_at)
    };
  }
};
var PersonNotFoundError = class extends Error {
};

// src/schema/base/fingerprints.ts
var FingerprintT = {
  deserialize: (data) => ({
    fingerprintId: data.fingerprint_id
  })
};
var FingerprintNotFoundError = class extends Error {
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AgentNotFoundError,
  AgentT,
  AgentVersionNotFoundError,
  AgentVersionT,
  FingerprintNotFoundError,
  FingerprintT,
  PersonNotFoundError,
  PersonT
});
